src/dashboard/src/components/administration/views_processing.py

```diff
@@ -47,26 +47,51 @@ def list(request):
 
 
 def edit(request, name=None):
-    if request.method == 'POST':
-        form = ProcessingConfigurationForm(request.POST)
-        if form.is_valid():
-            try:
-                form.save_config()
-            except Exception:
-                msg = 'Failed to save processing config.'
-                logger.exception(msg)
-                messages.error(request, msg)
-            else:
-                messages.info(request, 'Saved!')
-            return redirect('components.administration.views_processing.list')
-    else:
-        form = ProcessingConfigurationForm()
-        if name is not None:
-            try:
-                form.load_config(name)
-            except IOError:
-                raise Http404
-    return render(request, 'administration/processing_edit.html', {'form': form})
+
+    def _redirect_list(msg=None):
+        if msg:
+            messages.info(request, msg)
+        return redirect(
+            'components.administration.views_processing.list')
+
+    def _report_error(error=None, error_msg=None):
+        if error is not None:
+            logger.exception("{} {}".format(error_msg, err))
+            messages.error(request, error_msg)
+        return redirect(
+            'components.administration.views_processing.list')
+
+    def _render_form():
+        return render(
+            request, 'administration/processing_edit.html',
+            {'form': form})
+
+    # Initialize form.
+    try:
+        form = ProcessingConfigurationForm(request.POST or None)
+    except Exception as err:
+        return _report_error(
+            err, "Unable to load processing configuration page.")
```

Comment:
It's important to capture potential errors here because the constructor of ProcessingConfigurationForm is going to communicate with MCPServer so we should really expect errors to occur at that point.